packages/ag-charts-community/src/chart/chart.test.ts

```diff
@@ -475,4 +464,50 @@ describe('Chart', () => {
             });
         });
     });
+
+    describe('Chart data inherited by Series', () => {
+        async function createChart(options: object) {
+            const chartOptions = prepareTestOptions(options);
+            const chartProxy = AgChart.create(chartOptions);
+            const chart = deproxy(chartProxy);
+            await waitForChartStability(chart);
+            return { chart, chartProxy, chartOptions };
+        }
+
+        it('Chart data inherited only when Series data is not defined ', async () => {
+            const moreData = datasets.economy.data;
+            const lessData = datasets.economy.data.slice(0, 2);
+            const { chart } = await createChart({
+                data: moreData,
+                series: [
+                    {
+                        type: 'bar',
+                        xKey: datasets.economy.categoryKey,
+                        yKey: datasets.economy.valueKey,
+                    },
+                    {
+                        type: 'bar',
+                        data: lessData,
+                        xKey: datasets.economy.categoryKey,
+                        yKey: datasets.economy.valueKey2,
+                    },
+                ],
+            });
+            expect(chart.data).toEqual(moreData);
+            expect(chart.series[0].data).toEqual(moreData);
+            expect(chart.series[1].data).toEqual(lessData);
+
+            chart.series[0].data = lessData;
+            chart.series[1].data = null;
```

Comment:
Rather than manually manipulating the implementation, could we instead use `AgChart.update()` to adjust the options appropriately?